### PR TITLE
Add tool tip formatting for time series charts

### PR DESCRIPTION
**Pull request by CharukaK**
## Purpose
Give the user to define the format of the time displayed in the tool tip in the chart configuration

## Aproach
introduce a new attribute in the chart configuration 'tipTimeFormat' and for it's value user should give a regex pattern that's defined in d3 [timeFormat](https://github.com/d3/d3-time-format/blob/master/README.md#locale_format) function

## Test environment
NPM v5.3.0, Node v8.5.0

### CODE REVIEW

File: src/components/ComponentGenerator.jsx [hunk 1]
```diff
@@ -73,8 +74,24 @@ export function getLineOrAreaComponent(config, chartIndex, onClick) {
...
+                            };
+                        } else {
+                            return (d) => {
+                                if (isNaN(d.x)) {
+                                    return `${config.x}:${d.x}\n${config.charts[chartIndex].y}:${Number(d.y).toFixed(2)}`;
+                                } else {
+                                    return `${config.x}:${Number(d.x).toFixed(2)}\n
+                                    ${config.charts[chartIndex].y}:${Number(d.y).toFixed(2)}`;
+                                }
+                                //                 `${config.x}:${Number(d.x).toFixed(2)}\n
```
Comment (TanyaM) [inline src/components/ComponentGenerator.jsx]: remove commented code

File: src/components/ComponentGenerator.jsx [hunk 2]
```diff
@@ -117,11 +134,30 @@ export function getLineOrAreaComponent(config, chartIndex, onClick) {
...
+                    } else {
+                        return (d) => {
+                            if (isNaN(d.x)) {
+                                return `${config.x}:${d.x}\n${config.charts[chartIndex].y}:${Number(d.y).toFixed(2)}`;
+                            } else {
+                                return `${config.x}:${Number(d.x).toFixed(2)}\n
+                                ${config.charts[chartIndex].y}:${Number(d.y).toFixed(2)}`;
+                            }
+                            //                 `${config.x}:${Number(d.x).toFixed(2)}\n
+                            // ${config.charts[chartIndex].y}:${Number(d.y).toFixed(2)}`;
```
Comment (TanyaM) [inline src/components/ComponentGenerator.jsx]: remove commented code

Comment (CharukaK) [reply src/components/ComponentGenerator.jsx]: done

File: src/components/ComponentGenerator.jsx (hunk 1 above)
Comment (CharukaK) [reply src/components/ComponentGenerator.jsx]: done

**Pull request merged**